library/std/src/keyword_docs.rs

```diff
@@ -1228,6 +1228,69 @@ mod ref_keyword {}
 /// ```
 mod return_keyword {}
 
+#[doc(keyword = "become")]
+//
+/// Perform a tail-call of a function.
+///
+/// A `become` transfers the execution flow to a function in such a way, that
+/// returning from the callee returns to the caller of the current function:
+///
+/// ```
+/// #![feature(explicit_tail_calls)]
+///
+/// fn a() -> u32 {
+///     become b();
+/// }
+///
+/// fn b() -> u32 {
+///     return 2; // this return directly returns to the main ---+
+/// }                  //                                        |
+///                    //                                        |
+/// fn main() {        //                                        |
+///     let res = a(); // <--------------------------------------+
+///     assert_eq!(res, 2);
+/// }
+/// ```
+///
+/// This is an optimization that allows function calls to not exhaust the stack.
+/// This is most useful for (mutually) recursive algorithms, but may be used in
+/// other cases too.
+///
+/// It is guaranteed that the call will not cause unbounded stack growth if it
+/// is part of a recursive cycle in the call graph.
+///
+/// For example note that the functions `halt` and `halt_loop` below are
+/// identical, they both do nothing, forever. However, `stack_overflow` is
+/// different from them, even though it is written almost identically to
+/// `halt`, `stack_overflow` exhausts the stack and so causes a stack
+/// overflow, instead of running forever.
```

Comment:
Isn't LLVM allowed to optimize `stack_overflow()` into `loop { }`? I know we don't allow it to remove infinite loops, but...

Reply:
It is allowed, but it also is allowed not to.

Reply:
Yeah. I guess it's allowed to do it in all these cases, right?

I guess what I'm concerned about is the example being so trivial that it doesn't hold up to even trivial examination.